lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1570,8 +1570,8 @@ class ShellCommand : public ExternalCommand {
       return false;
     }
     
-    // Collect the discovered dependencies, if used.
-    if (!depsPaths.empty()) {
+    // Collect the discovered dependencies, if used and command was not cancelled.
+    if (!depsPaths.empty() && !bsci.getDelegate().isCancelled()) {
```

Comment:
I see what's happening now: https://github.com/apple/swift-llbuild/blob/master/lib/BuildSystem/LaneBasedExecutionQueue.cpp#L470 -- these are commands that have been started but were signalled during their execution, those currently return `true`. I think the right thing to do is to indeed create an enum so that we can act more appropriately on cancellation.